Eatery/Views/Eateries/PopularTimesView.swift

```diff
@@ -0,0 +1,205 @@
+
+        setExpanded(isExpanded, animated: false)
+        histogramView.reloadData()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    @objc private func showHideButtonPressed(_ sender: UIButton) {
+        setExpanded(!isExpanded, animated: true)
+    }
+
+    func setExpanded(_ newValue: Bool, animated: Bool) {
+        isExpanded = newValue
+        if isExpanded {
+            showHideButton.setTitle("Hide", for: .normal)
+        } else {
+            showHideButton.setTitle("Show", for: .normal)
+        }
+
+        let actions: (() -> Void) = {
+            self.histogramView.alpha = self.isExpanded ? 1.0 : 0.0
+
+            if self.isExpanded {
+                self.histogramCollapseConstraint?.deactivate()
+                self.histogramExpandConstraint?.activate()
+            } else {
+                self.histogramExpandConstraint?.deactivate()
+                self.histogramCollapseConstraint?.activate()
+            }
+
+            self.layoutDelegate?.popularTimesContentSizeDidChange(self)
+        }
+
+        if animated {
+            UIViewPropertyAnimator(duration: 0.5, dampingRatio: 1.0, animations: actions).startAnimation()
+        } else {
+            actions()
+        }
+    }
+
+}
+
+extension PopularTimesView: HistogramViewDataSource {
+
+    // Overflowed military hour is a time between 0...47
+    private func containOverflowedMilitaryHour(hour: Int) -> Int {
+        return (hour >= 24) ? hour - 24 : hour
+    }
+
+    private func formattedHourForTime(militaryHour: Int) -> String {
+        var formattedHour: String!
+        if militaryHour > 12 {
+            formattedHour = "\(militaryHour - 12)p"
+        } else if militaryHour == 12 {
+            formattedHour = "12p"
+        } else if militaryHour > 0 {
+            formattedHour = "\(militaryHour)a"
+        } else {
+            formattedHour = "12a"
+        }
+        return formattedHour
+    }
+
+    func numberOfDataPoints(for histogramView: HistogramView) -> Int {
+        return overflowedEndHour - startHour
+    }
+
+    func histogramView(_ histogramView: HistogramView, relativeValueOfDataPointAt index: Int) -> Double {
+        return eatery.swipeDensity(for: containOverflowedMilitaryHour(hour: startHour + index))
+    }
+
+    func histogramView(_ histogramView: HistogramView, descriptionForDataPointAt index: Int) -> NSAttributedString? {
+        let currentHour = Calendar.current.component(.hour, from: Date())
+        let barHour = containOverflowedMilitaryHour(hour: startHour + index)
+
+        let hourText = currentHour == barHour ? "Now: " : formattedHourForTime(militaryHour: barHour).appending(": ")
+
+        let waitTimesText: String
+        if let (low: lowEstimate, high: highEstimate) = eatery.waitTimes(atHour: barHour, minute: 0) {
+            waitTimesText = "\(lowEstimate)-\(highEstimate)m"
+        } else {
+            waitTimesText = "?"
```

Comment:
not sure if this should be `?`, does design have a state for this?

Reply:
I'll ask